### PR TITLE
Fixed some wrapper warnings

### PR DESCRIPTION
**Pull request by varunagrawal**
Due to

- Eigen alignment.
- Wrapper code being too long.
- Exceptions which weren't reference.

---
This change is [Reviewable](https://reviewable.io/reviews/borglab/gtsam/260)

### CODE REVIEW

File: gtsam/nonlinear/utilities.h [hunk 1]
```diff
@@ -237,12 +237,12 @@ Values localToWorld(const Values& local, const Pose2& base,
...
       Pose2 pose = local.at<Pose2>(key);
       world.insert(key, base.compose(pose));
-    } catch (std::exception e1) {
+    } catch (const std::exception& e1) {
       try {
         // if value is a Point2, transform it from base pose
         Point2 point = local.at<Point2>(key);
         world.insert(key, base.transformFrom(point));
-      } catch (std::exception e2) {
+      } catch (const std::exception& e2) {
```
Comment (ProfFan) [inline gtsam/nonlinear/utilities.h]: Can you add a debug output here if we are in debug mode?

Comment (varunagrawal) [reply gtsam/nonlinear/utilities.h]: Absolutely

Comment (varunagrawal) [reply gtsam/nonlinear/utilities.h]: Done

**Comment by varunagrawal** [comment]
@ProfFan one last review and if you think this looks good, I can merge it in.

**Pull request merged**